### PR TITLE
fix testcase, add loop param

### PR DESCRIPTION
**Pull request by smurfix**
Two changes:
* a testcase had a missing @testing.coroutine decorator.
* add a loop parameter so that it's possible to run aioamqp in another thread.

### CODE REVIEW

**Comment by dzen** [comment]
Hello @smurfix ! Thank  you for your PR.

Would you please add a test with a connection using another EventLoop ?

Thank you !

**Comment by smurfix** [comment]
Not a problem. Simply remove the code to set the event loop in testing.py and test_protocol.py, and pass self.loop wherever necessary.
Any spot where the original mainloop is would result in a locked testcase. (Nothing that can be done about that.)

I just pushed that: 89f8a1b
and also b5c362e which fixes a typo ("hearbeat"?)

**Comment by smurfix** [comment]
Re-testing with ```PYTHONASYNCIODEBUG=1```

**Comment by smurfix** [comment]
rebased as 7821edb, should work now

File: aioamqp/__init__.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @asyncio.coroutine
 def connect(host='localhost', port=None, login='guest', password='guest',
             virtualhost='/', ssl=False, login_method='AMQPLAIN', insist=False,
-            protocol_factory=AmqpProtocol, *, verify_ssl=True, **kwargs):
+            protocol_factory=AmqpProtocol, *, verify_ssl=True, loop=None, **kwargs):
```
Comment (rbarrois) [inline aioamqp/__init__.py]: The ``loop`` argument should be described in the docstring.

File: aioamqp/protocol.py [hunk 2]
```diff
@@ -25,15 +25,15 @@ class AmqpProtocol(asyncio.StreamReaderProtocol):
     CHANNEL_FACTORY = amqp_channel.Channel
 
     def __init__(self, *args, **kwargs):
-        loop = kwargs.get('loop') or asyncio.get_event_loop()
-        super().__init__(asyncio.StreamReader(loop=loop), self.client_connected, loop=loop)
+        self._loop = kwargs.get('loop') or asyncio.get_event_loop()
```
Comment (rbarrois) [inline aioamqp/protocol.py]: Since this attribute is used by other classes, shouldn't it be a "public" attribute, i.e ``self.loop``?

**Comment by rbarrois** [comment]
@smurfix this looks great!
However, would you mind adding a small note on this new option in the docs?
Likely in the ``api.rst`` section?

Thanks :)

**Comment by dzen** [comment]
I'll add a new entry in the changlog section : https://github.com/Polyconseil/aioamqp/blob/master/docs/changelog.rst

File: aioamqp/protocol.py (hunk 2 above)
Comment (smurfix) [inline aioamqp/protocol.py]: It's not. All uses explicitly pass self._loop to the other class's constructor.

**Comment by smurfix** [comment]
Pushed doc change(s).

**Comment by dzen** [comment]
Thank you ! I pushed your changes just now !

**Pull request merged**